Features/gcp taar amodump

crankycoder opened the pull request:
These are patches to port the taar_amodump DAG to use GCP.

The first few jobs do not require Spark:

  * taar_amodump
  * taar_amo_whitelist
  * taar_update_whitelist

The last job requires PySpark:

  * taar_lite 

Of note - 3 Airflow connections:

GCP connections: `google_cloud_derived_datasets_3` and `google_cloud_airflow_dataproc`.

AWS connection: `airflow_taar_rw_s3`

This PR requires a patch to https://github.com/mozilla/taar_gcp_etl/ so that the taar_lite_guidguid.py script is pushed into a GCS bucket on a tagged build.

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
+from operators.gcp_container_operator import GKEPodOperator  # noqa
+from utils.dataproc import moz_dataproc_pyspark_runner
+
+# Unsalted cluster name so subsequent runs fail if the cluster name exists
```
Rm comment. This is a long lived cluster

crankycoder replied:
Comment deleted.

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
+# Unsalted cluster name so subsequent runs fail if the cluster name exists
+cluster_name = "bq-load-gke-1"
+
+taarlite_cluster_name = "taarlite-guidguid"
```
Maybe add dataproc to the name

crankycoder replied:
renamed to `dataproc-taarlite-guidguid`

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
 
-from utils.mozetl import mozetl_envvar
+# Defined in Airflow's UI -> Admin -> Connections
+gcp_conn_id = "google_cloud_derived_datasets_3"
```
Rm the _3

crankycoder replied:
dropped the `_3`

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
+    cluster_name="bq-load-gke-1",
+    name="taar-amodump",
+    namespace="default",
+    image="gcr.io/moz-fx-data-airflow-prod-88e0/taar_gcp_etl:0.1",
```
Comment to link to external github repo here would be nice

crankycoder replied:
I've added comments to each of the jobs to point to the repo now

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
-    output_visibility="private",
-    dag=dag
+    email=["mlopatka@mozilla.com", "vng@mozilla.com", "hwoo@mozilla.com"],
+    arguments=["-m", "taar_etl.taar_amowhitelist"],
```
No date req?

crankycoder replied:
Nope. But added comments about it.  

The job just crawls the AMO API site to build a list of available addons.  Adding a date parameter wouldn't give us a real date scoped list of addons as AMO is always going to give us the latest available addon list.

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,122 @@
+        default_args=default_args,
+        cluster_name=taarlite_cluster_name,
+        job_name="TAAR_Lite_GUID_GUID",
+        python_driver_code="gs://temp-hwoo-removemelater/taar_lite_guidguid.py",
```
This will need to change prior to merge

crankycoder replied:
@haroldwoo Can you help me get this integrated into CircleCI so that tagged builds automatically push to a GCS bucket?

haroldwoo replied:
as long as the script doesnt have any secrets, you can put a copy of the script in this repo under the jobs folder, and itll be sync'd to gs://moz-fx-data-prod-airflow-dataproc-artifacts/jobs after this PR merges.

https://github.com/mozilla/telemetry-airflow/blob/master/.circleci/config.yml#L74

crankycoder replied:
Thanks - I've put a copy of the script into the data directory in 09d9487

crankycoder commented:
@haroldwoo r?  I've put the taar_lite_guidguid script into the jobs directory per your last suggestion.

crankycoder commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,127 @@
+        cluster_name=taarlite_cluster_name,
+        job_name="TAAR_Lite_GUID_GUID",
+        python_driver_code="gs://moz-fx-data-prod-airflow-dataproc-artifacts/jobs/taar_lite_guidguid.py",
+        num_workers=15,
```
@haroldwoo @mreid-moz is this ok?  I bumped the number of workers up to 15 so that the job completes in ~30 minutes.  Is there a cost optimization that I should think about?  This job needs to run daily.

haroldwoo replied:
Should be fine. Does this job take roughly twice the time with twice as many workers?

haroldwoo commented on dags/taar_amodump.py:
```diff
@@ -1,83 +1,127 @@
+from operators.gcp_container_operator import GKEPodOperator  # noqa
+from utils.dataproc import moz_dataproc_pyspark_runner
+
+cluster_name = "bq-load-gke-1"
```
I would rename this variable to something like gke_cluster_name to avoid confusion. Then use the variable on lines 44, 64, 86

crankycoder commented:
This DAG runs successfully for me on a local airflow instance.

![Airflow_-_DAGs](https://user-images.githubusercontent.com/4829/66600995-d394f880-eb74-11e9-8e8b-9081b8e6a972.jpg)

I've squashed and rebased to clarify the changes.  I've left the switch from test GCS and production GCS file paths in df629ae and 24f572e as I can't test the production GCS path on my local airflow instance.

crankycoder commented:
@haroldwoo 33d99b0a explicitly removes the `google_cloud_derived_datasets` connection as it causes surprising failures during test as the connection is defined with empty values for all connection parameters.
